code/datums/elements/_element.dm

```diff
@@ -31,12 +37,15 @@
 
 /// Finds the singleton for the element type given and attaches it to src
 /datum/proc/AddElement(eletype, ...)
-	var/datum/element/ele = SSdcs.GetElement(eletype)
+	var/datum/element/ele = SSdcs.GetElement(arglist(args))
 	args[1] = src
 	if(ele.Attach(arglist(args)) == ELEMENT_INCOMPATIBLE)
 		CRASH("Incompatible [eletype] assigned to a [type]! args: [json_encode(args)]")
 
-/// Finds the singleton for the element type given and detaches it from src
-/datum/proc/RemoveElement(eletype)
-	var/datum/element/ele = SSdcs.GetElement(eletype)
+/**
+  * Finds the singleton for the element type given and detaches it from src
+  * You only need additional arguments beyond the type if you're using ELEMENT_BESPOKE
+  */
+/datum/proc/RemoveElement(eletype, ...)
```

Comment:
Shouldn't there be a way to remove these without knowing the arguments passed.

Reply:
The problem is there could be multiple of the same type of element but with different vars and with only the type the code wouldn't know which you want to remove. Especially if you're using this feature to apply multiple variations of the element to one thing. Best I could do is make AddElement return a key which you could use to remove it later